src/client.ts

```diff
@@ -110,37 +116,38 @@ export class SubscriptionClient {
     this.client.close();
   }
 
-  public subscribe(options: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
-    const { query, variables, operationName, context } = options;
+  public subscribe(opts: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
+    // this.eventEmitter.emit('subscribe', options);
+    this.checkSubscriptionParams(opts, handler);
```

Comment:
so if we change this, it will break the following tests as they currently stand:

1. should throw an exception when query is not provided
2. should throw an exception when query is not valid

because those errors currently throw outside the handler

i'm all for deferring all error handling to the handler (aside from throwing if there isn't a handler), but was trying to make this feature without making breaking changes. but i agree it's cleaner -- so if you confirm this is the route to go, i'm happy to make the change